games/armagetronad/Portfile

```
@@ -0,0 +1,291 @@
+        foreach re [list \
+            "/CONFIG=.*sysconfdir/a\\
+                \[ X\"\${DATADIR}\" = X\"\" \] && DATADIR=@aa_datadir@\\
+            " \
+            "s/\(--userdatadir\)/--datadir \${DATADIR} \\1/" \
+        ] {
+            reinplace -E $re ${worksrcpath}/batch/serverstarter.in
+        }
+
+        # Avoid runtime error "nohup: can't detach from console:
+        # Inappropriate ioctl for device". (Reference:
+        # https://stackoverflow.com/questions/29112446/nohup-doesnt-work-with-os-x-yosmite-get-error-cant-detach-from-console-no-s)
+        reinplace {s/nohup //} ${worksrcpath}/batch/rcd_startstop.in
+    }
+}
+
+configure.args-append       --enable-music \
+                            --disable-useradd \
+                            --disable-uninstall \
+                            --disable-games
+if {${os.platform} eq "darwin"} {
+    configure.args-append   --disable-etc
+}
+if {$subport ne ${name}} {
+    configure.args-append   --disable-desktop
+}
+if {$subport eq "${name}-dedicated"} {
+    configure.args-append   --enable-dedicated
+    if {${os.platform} eq "darwin"} {
+        configure.args-append   --disable-initscripts
+    }
+}
+
+if {$subport eq "${name}-dedicated"} {
+    if {${os.platform} eq "darwin" && ${os.major} >= 19} {
+        set usershell /bin/zsh
+    } else {
+        set usershell /bin/bash
+    }
+    add_users   ${name} group=${name} shell=$usershell \
+                realname=[shellescape $pretty_name]
+
+    startupitem.create      yes
+    startupitem.autostart   no
+    startupitem.name        $subport
+    foreach action [list start stop restart] {
+        startupitem.$action \
+            "${prefix}/share/$subport/scripts/rcd_server $action"
+    }
+    set vardir  ${prefix}/var
+    startupitem.pidfile     auto $vardir/run/$subport/${subport}.pid
+    startupitem.logfile     $vardir/log/$subport/${name}.log
+    startupitem.logevents   yes
+}
+
+if {$subport eq ${name}} {
+    PortGroup               app 1.0
+
+    # For names with spaces, we have to expand the variable in order to
+    # use it for app.name. See: https://trac.macports.org/ticket/64326
+    app.name                {*}$pretty_name
+    app.icon                MacOS/${pretty_name}.icns
+    app.retina              yes
+    app.use_launch_script   yes
+
+    post-destroot {
+        set app_contents ${applications_dir}/${pretty_name}.app/Contents
+        set app_resources $app_contents/Resources
+        copy ${worksrcpath}/src/macosx/English.lproj \
+             ${destroot}$app_resources/
+        set infoplist_strings \
+            $app_resources/English.lproj/InfoPlist.strings
+        move ${destroot}${infoplist_strings}.in \
+             ${destroot}${infoplist_strings}
+        foreach re [list \
+            "s/@version@/${version}/" \
+            "s/@year@/$year_released/" \
+        ] {
+            reinplace $re ${destroot}$infoplist_strings
+        }
```

This can be done as a one-liner... and usually when we replace placeholders we use the `g` flag to ensure all occurrences are replaced.
```suggestion
        reinplace "s/@version@/${version}/g;s/@year@/$year_released/g" ${destroot}$infoplist_strings
```

---

Ah nice catch. It's always good to be able to eliminate a for-loop. 👍